File: 2299-merge-nodes-in-between-zeros/merge-nodes-in-between-zeros.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def mergeNodes(self, head: Optional[ListNode]) -> Optional[ListNode]:
        stack = []
        node = head.next
        nextZero = False
        while node.next:
           
            # while node.next.val !=0 and node.next:
            if node.next.val==0:
                if not node.next.next:
                    node.next = node.next.next
                    return head.next
                # print("at 0",node)
                # print()
                node = node.next
            # print("at node:",node.val,node)
            if node.next:
                node.val += node.next.val
                node.next = node.next.next
                # print("now node is:",node.val,node.next)

        

File: 2299-merge-nodes-in-between-zeros/merge-nodes-in-between-zeros.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */

// void print(ListNode* head) {
//     ListNode* current = head;
//     while (current != nullptr) {
//         std::cout << current->val;
//         if (current->next != nullptr) {
//             std::cout << " -> ";
//         }
//         current = current->next;
//     }
//     std::cout << std::endl;
// }


class Solution {
public:
    ListNode* mergeNodes(ListNode* head) {
        ListNode* current = head->next;
        while (current->next != nullptr) {
            if (current->next->val == 0) {

                if (current->next->next == nullptr){
                    current->next= nullptr;
                    return head->next;
                }

                current = current->next;
            }

            if (current->next != nullptr){
                current->val += current->next->val;
                current->next = current->next->next;
            }
        }


        // print(head);
        return head->next;
        
    }
};